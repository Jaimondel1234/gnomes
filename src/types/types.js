export const types = {
  login: "[Auth] Login",
  logout: "[Auth] Logout",
  loginRequest: "[Auth] Login Request",
};
