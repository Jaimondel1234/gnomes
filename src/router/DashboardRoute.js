import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { Navbar } from '../components/ui/Navbar';
import { HomeScreen } from '../components/home/HomeScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { GnomeScreen } from '../components/GnomeList/GnomeScreen';
import { GnomeDetailScreen } from '../components/GnomeDetail/GnomeDetailScreen';

/* 
  Routes if logged
*/
export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <Switch>
        <Route exact path='/home' component={HomeScreen} />
        <Route exact path='/search' component={SearchScreen} />
        <Route exact path='/all-gnomes' component={GnomeScreen} />
        <Route exact path='/gnome-detail' component={GnomeDetailScreen} />
        <Redirect to='/home' />
      </Switch>
    </>
  );
};
