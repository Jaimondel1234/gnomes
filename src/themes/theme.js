/* 
  Theme
*/

export const theme = {
  color: {
    main: '#007bff',
  },
};
