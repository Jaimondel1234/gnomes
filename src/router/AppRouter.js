import React, { useState, useEffect } from 'react';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { LoginScreen } from '../auth/LoginScreen';
import { DashboardRoutes } from './DashboardRoute';
import { useSelector, useDispatch } from 'react-redux';
import { login } from '../actions/auth';

/* AppRouter (root)
 */

export function AppRouter() {
  const dispatch = useDispatch();
  const { uid } = useSelector((state) => state.auth);
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  useEffect(() => {
    const user = JSON.parse(localStorage.getItem('user'));
    if (user) {
      dispatch(login(user.uid, user.displayName));
    }
  }, [dispatch, localStorage, login]);

  useEffect(() => {
    setIsAuthenticated(!!uid);
  }, [uid, setIsAuthenticated]);

  return (
    <Router>
      <Switch>
        <PublicRoute
          exact
          path='/login'
          component={LoginScreen}
          isAuthenticated={isAuthenticated}
        />
        <PrivateRoute
          path='/'
          component={DashboardRoutes}
          isAuthenticated={isAuthenticated}
        />
        <Redirect to='/' />
      </Switch>
    </Router>
  );
}
