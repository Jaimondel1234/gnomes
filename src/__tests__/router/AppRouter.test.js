import React from "react";

import { AppRouter } from "../../router/AppRouter";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { theme } from "../../themes/theme";
import { types } from "../../types/types";

const user = { uid: 23344, displayName: "Fer" };
localStorage.setItem("user", JSON.stringify(user));

describe("Test <AppRouter/>", () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initState = {
    auth: {},
  };

  const store = mockStore(initState);
  store.dispatch = jest.fn();

  test("it must be displayed correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <AppRouter />
        </ThemeProvider>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("if logged in, user information must be stored in redux", () => {
    expect(store.dispatch).toHaveBeenCalledWith({
      type: types.login,
      displayName: "Fer",
      uid: 23344,
    });
  });
});
