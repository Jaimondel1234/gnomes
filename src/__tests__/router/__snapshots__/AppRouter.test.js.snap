// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AppRouter/> it must be displayed correctly 1`] = `
<div
  className="core__PanelStyle-qu5lcc-1 jqDqbg"
  color="#ffff"
>
  <div
    className="core__TextStyle-qu5lcc-3 ccva-db"
    fontSize="60px"
    fontWeight="bold"
  >
    Login
  </div>
  <form
    className="core__FlexStyle-qu5lcc-4 joXSZK"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="core__InputStyle-qu5lcc-5 LoginScreen__LoginInputStyle-sc-1or96zl-0 eMTzQF dhIvMS"
      name="email"
      onChange={[Function]}
      placeholder="Your email"
      type="text"
      value="heroe@gmail.com"
    />
    <input
      autoComplete="off"
      className="core__InputStyle-qu5lcc-5 LoginScreen__LoginInputStyle-sc-1or96zl-0 eMTzQF dhIvMS"
      name="password"
      onChange={[Function]}
      placeholder="Your password"
      type="password"
      value="heroe"
    />
    <button
      className="core__ButtonStyle-qu5lcc-6 gzLaSK"
      fontSize="25px"
      fontWeight="bold"
      letterSpacing="3px"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
